### ISSUE TITLE
Invalid MinimumOSVersion when submitting app to App Store Connect.

### ISSUE THREAD

**Issue by bullinnyc** [c-villain/YandexMapsMobile#11]
**Description:**

I’m encountering an issue with the YandexMapsMobile framework when trying to submit my app to App Store Connect. The error states that the MinimumOSVersion key is missing or invalid in the framework’s Info.plist. This issue only occurs when submitting the app the project builds and runs fine locally.

**Error Details:**

Here are the error messages received during the submission:  

-   Invalid MinimumOSVersion. Apps that only support 64-bit devices must specify a deployment target of 8.0 or later. 

- Missing Info.plist value. A value for the key 'MinimumOSVersion' in bundle MyApp.app/Frameworks/YandexMapsMobile.framework is required. 

- Invalid Bundle. The bundle MyApp.app/Frameworks/YandexMapsMobile.framework does not support the minimum OS Version specified in the Info.plist.

**Steps to Reproduce:**

1. Install YandexMapsMobile using Swift Package Manager.
2. Build the app in Xcode (no issues).
3. Attempt to distribute the app through App Store Connect.
4. The above errors occur during the validation step.

**Environment:**

- Xcode Version: 15.4
- Swift Version: 5
- Deployment Target: iOS 15.0
- YandexMapsMobile Version: 4.7.0
- Installation Method: Swift Package Manager

**What I Tried:**

I added a script to modify the Info.plist file of the YandexMapsMobile framework to include a MinimumOSVersion. While this works as a temporary workaround, it feels like a hack and shouldn’t be necessary.

**Expected Behavior:**

The framework should have a proper MinimumOSVersion set in its Info.plist to comply with App Store Connect requirements.

**Actual Behavior:**

The MinimumOSVersion is missing or improperly set, causing validation errors when submitting to App Store Connect.

**Proposed Solution:**

Ensure that the framework’s Info.plist contains a valid MinimumOSVersion key with a value.

**Additional Context:**

This issue seems to be specific to the App Store submission process, as local builds do not encounter this error.  

Please let me know if you need any additional information or steps to reproduce the issue.

**Comment by c-villain** [comment]
@bullinnyc Hello! Thx for reporting 🤝 Have u read [this one](https://github.com/c-villain/YandexMapsMobile/issues/5)?

P.S. Will u share the script

**Comment by bullinnyc** [comment]
> @bullinnyc Hello! Thx for reporting 🤝 Have u read [this one](https://github.com/c-villain/YandexMapsMobile/issues/5)?

Yes, I encountered such a problem and used the solution to this issues.  
But the problem of submit my app to App Store Connect is not related to this.

> P.S. Will u share the script

Yes, sure.  

<img width="970" alt="screen" src="https://github.com/user-attachments/assets/4ae5910e-da41-4352-a097-795e016b67f3">  

```
FRAMEWORK_PATH="${BUILT_PRODUCTS_DIR}/${FRAMEWORKS_FOLDER_PATH}/YandexMapsMobile.framework"
PLIST_PATH="${FRAMEWORK_PATH}/Info.plist"
MIN_OS_VERSION="15.0"

if [[ -f "$PLIST_PATH" ]]; then
    if /usr/libexec/PlistBuddy -c "Print :MinimumOSVersion" "$PLIST_PATH" 2>/dev/null; then
        echo "MinimumOSVersion already exists."
    else
        echo "Adding MinimumOSVersion."
        /usr/libexec/PlistBuddy -c "Add :MinimumOSVersion string $MIN_OS_VERSION" "$PLIST_PATH"
    fi
else
    echo "Info.plist not found on path: $PLIST_PATH"
fi
```

**Comment by KonkeyDong37** [comment]
Same problem in Xcode 16

Target: iOS 15.0
YandexMapsMobile Version: 4.8.0

**Comment by c-villain** [comment]
@KonkeyDong37 Did u add DeviceCheck to Other Linker Flags?

**Comment by KonkeyDong37** [comment]
Yes, I checked against the list in the "Recommedations to use" section and it all matches up

**Comment by c-villain** [comment]
@KonkeyDong37 try to add @bullinnyc script 🤷🏼‍♂️

**Comment by KonkeyDong37** [comment]
Already tried it, this script removes 2 errors out of 3. One remains

Invalid Bundle. The bundle XXXX.app/Frameworks/YandexMapsMobile.framework does not support the minimum OS Version specified in the Info.plist. (ID: f7f64e37-41b1-492b-8d87-c7fc67183f09)

**Comment by bullinnyc** [comment]
> Already tried it, this script removes 2 errors out of 3. One remains
> 
> Invalid Bundle. The bundle XXXX.app/Frameworks/YandexMapsMobile.framework does not support the minimum OS Version specified in the Info.plist. (ID: f7f64e37-41b1-492b-8d87-c7fc67183f09)

The script still works for me. And at Xcode16 too. It is important that `MIN_OS_VERSION` in the script matches the `Minimum Deployments` target in the project, otherwise this error remains.

**Comment by bullinnyc** [comment]
@c-villain We made a test SPM package, changed the Info.plist files, there was an error in the `MinimumOSVersion` register, after that submit to App Store Connect was successful.  

<img width="1242" alt="screenshot" src="https://github.com/user-attachments/assets/75803f32-b55f-46c7-860d-fd9b82b0d428">

**Comment by c-villain** [comment]
@bullinnyc Thanks for your note! Yes, in plist files the case of keys matters. Using an incorrect case may result in the system not recognizing the key. 

@KonkeyDong37 @bullinnyc I've just  updated the whole release with 4.8.0 tag. Can you check it with proceeding? Before doing this you should clear caches of your spm in Xcode project otherwise Xcode won't detect the new version of spm.

**Comment by bullinnyc** [comment]
> @bullinnyc Thanks for your note! Yes, in plist files the case of keys matters. Using an incorrect case may result in the system not recognizing the key.
> 
> @KonkeyDong37 @bullinnyc I've just updated the whole release with 4.8.0 tag. Can you check it with proceeding? Before doing this you should clear caches of your spm in Xcode project otherwise Xcode won't detect the new version of spm.

There is only one error left:

- Invalid Bundle. The bundle MyApp.app/Frameworks/YandexMapsMobile.framework does not support the minimum OS Version specified in the Info.plist.  

In the test version, we removed the `MinimumOSVersion` key from the root Info.plist file.  

<img width="365" alt="screenshot" src="https://github.com/user-attachments/assets/3064c38d-a710-4b6f-8c1c-74eb114cfd3a">
<img width="1042" alt="screenshot-2" src="https://github.com/user-attachments/assets/3d6bcddf-e8e4-478f-930a-673de0b7b227">

**Comment by bullinnyc** [comment]
> @bullinnyc
> 
> ok, and what will be if minimum OS version tag still remains in root plist? The error "Invalid bundle"?

Yes, probably this last error remained because of the `MinimumOSVersion` key in the root Info.plist file, we did not change anything else.

**Comment by c-villain** [comment]
@bullinnyc updated release with last recommendations )) Could you check it with submitting?

**Comment by bullinnyc** [comment]
> @bullinnyc updated release with last recommendations )) Could you check it with submitting?

No, the error remains. I missed an important detail, we set `MinimumOSVersion` to 100.0.  

<img width="796" alt="screen" src="https://github.com/user-attachments/assets/02747b6f-6d64-4c64-be0b-33d51b92186b">  

Firebase uses a hack that just set the MinimumOSVersion to 100.0 to pass this new check.  

The solution seems to be to set the `MinimumOSVersion` for the framework to be greater than or equal to the app's `Minimum Deployments` version in project. It seems to be an Apple bug.

More info:
https://github.com/singular-labs/Singular-iOS-SDK/issues/55#issue-2189187046
https://github.com/firebase/firebase-ios-sdk/pull/12439#issuecomment-1994466302

**Comment by c-villain** [comment]
> More info: [singular-labs/Singular-iOS-SDK#55 (comment)](https://github.com/singular-labs/Singular-iOS-SDK/issues/55#issue-2189187046) 

@bullinnyc seems that singular-labs used this hack only for 4.3.3 version not for the others.

**Comment by bullinnyc** [comment]
> > More info: [singular-labs/Singular-iOS-SDK#55 (comment)](https://github.com/singular-labs/Singular-iOS-SDK/issues/55#issue-2189187046)
> 
> @bullinnyc seems that singular-labs used this hack only for 4.3.3 version not for the others.

12.3.3? Sounds like the truth.

**Comment by c-villain** [comment]
> 12.3.3? Sounds like the truth.

yes, 12.3.3

**Comment by FuzzzzyBoy** [comment]
I have same probliem  
So, what is the solution here?

**Comment by FuzzzzyBoy** [comment]
I have same problem
So, what is the solution here?

**Comment by FuzzzzyBoy** [comment]
I have same problem while distributing application to AppStore Connect

> Invalid Bundle. The bundle MyApp.app/Frameworks/YandexMapsMobile.framework does not support the minimum OS Version specified in the Info.plist.

So, what is the solution here? I am using your package as sub-package in my project

**Comment by c-villain** [comment]
@bullinnyc Just updated the release 4.8.0 Could you check it with submitting? Clean project before
